src/h5web/providers/hsds/utils.ts

```diff
@@ -1,5 +1,13 @@
+import { HDF5Collection } from '../hdf5-models';
+import { EntityKind } from '../models';
 import type { HsdsLink, HsdsExternalLink } from './models';
 
 export function isHsdsExternalLink(link: HsdsLink): link is HsdsExternalLink {
   return 'h5domain' in link;
 }
+
+export const COLLECTION_TO_KIND: Record<HDF5Collection, EntityKind> = {
+  [HDF5Collection.Groups]: EntityKind.Group,
+  [HDF5Collection.Datasets]: EntityKind.Dataset,
+  [HDF5Collection.Datatypes]: EntityKind.Datatype,
+};
```

Comment:
A bit awkward but there are no way around it I guess

Reply:
I've extracted this into a function in the new PR, but yeah, the HSDS provider code overall is pretty bad. Hopefully the situation will improve from HSDS' end eventually.